src/main/java/com/example/clone/controller/CommentController.java

```diff
@@ -0,0 +1,36 @@
+package com.example.clone.controller;
+
+import com.example.clone.model.Comment;
+import com.example.clone.repository.CommentRepository;
+import com.example.clone.repository.PostRepository;
+import com.example.clone.repository.UserInfoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import com.example.clone.service.CommentService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/comments")
+public class CommentController {
+
+@Autowired
+    private CommentService commentService;
+    @Autowired
+    private CommentRepository commentRepository;
+
+    @Autowired
+    private PostRepository postRepository;
+
+    @Autowired
+    private UserInfoRepository userInfoRepository;
```

Comment:
There are three types of autowiring in Spring:

- Inject by setter
- Inject by constructor
- Inject by field

Of those three constructor is preferable way because setter makes class mutable and because filed utilizes Java Reflection which is expensive operation performance-wise. 

https://www.quora.com/Is-Java-Reflection-slow-or-expensive